### PR TITLE
Feature 1098/validation for feedback request date selection

### PR DESCRIPTION
**Pull request by JEdwardBrown**

### CODE REVIEW

**Review by moshirim — APPROVED** [review]

**Comment by bhatiam12** [comment]
@smithj11 @JEdwardBrown 
builds are failing -
```./src/pages/FeedbackRequestPage.jsx
  Line 139:8:  React Hook useCallback has a missing dependency: 'handleSubmit'. Either include it or remove the dependency array  react-hooks/exhaustive-deps```

**Comment by JEdwardBrown** [comment]
Yep! Just some issues that are popping after fixing some merge conflicts, I'm working on them now.

**Comment by JEdwardBrown** [comment]
@bhatiam12 everything should be working fine now

**Review by sagetheherb — APPROVED** [review]

**Review by bhatiam12 — APPROVED** [review]

File: web-ui/src/pages/FeedbackRequestPage.jsx [hunk 1]
```diff
@@ -77,26 +81,35 @@ const FeedbackRequestPage = () => {
...
+      let timeStamp = Date.parse(dateString)
+      if(dateString < today)
+        return false;
+      else
+        return !isNaN(timeStamp);
+     }, []);
+
+    const hasSend = useCallback(() => {
+      let isValidPair = false
+      if(dueQuery && dueQuery !== undefined) {
```
Comment (pieperm) [inline web-ui/src/pages/FeedbackRequestPage.jsx:108]: This can be simplified to `if (dueQuery) {`

File: web-ui/src/pages/FeedbackRequestPage.jsx [hunk 2]
```diff
@@ -105,49 +118,48 @@ const FeedbackRequestPage = () => {
...
     history.push("/feedback/request/confirmation");
   }, [history]);
 
-  const onNextClick = useCallback(() => {
-    if (!canProceed()) return;
-    if (activeStep === steps.length) {
-      handleSubmit();
-      return;
+    if (activeStep < 1 || activeStep > steps.length) {
+
```
Comment (pieperm) [inline web-ui/src/pages/FeedbackRequestPage.jsx:131]: I don't think this block does anything?

File: web-ui/src/pages/FeedbackRequestPage.jsx [hunk 3]
```diff
@@ -77,26 +81,35 @@ const FeedbackRequestPage = () => {
...
+      else
+        return !isNaN(timeStamp);
+     }, []);
+
+    const hasSend = useCallback(() => {
+      let isValidPair = false
+      if(dueQuery && dueQuery !== undefined) {
+        isValidPair = dueQuery >= sendQuery
+      }
+      return (sendQuery && isValidDate(sendQuery) && isValidPair)
```
Comment (pieperm) [inline web-ui/src/pages/FeedbackRequestPage.jsx:111]: Would it make more sense to move the logic for `isValidPair` into the `isValidDate` function?

**Review by pieperm — CHANGES_REQUESTED** [review]

**Review by pieperm — APPROVED** [review]

**Pull request merged**